api/api/indices.js

```diff
@@ -1568,30 +1620,20 @@ IndicesApi.prototype.putMapping = function indicesPutMappingApi(params, options,
   let path = '';
   if (index != null && type != null) {
     if (method == null) method = 'PUT';
-    path = '/' + encodeURIComponent(index) + '/' + encodeURIComponent(type) + '/' + '_mapping';
-  } else if (index != null && type != null) {
-    if (method == null) method = 'PUT';
-    path = '/' + encodeURIComponent(index) + '/' + '_mapping' + '/' + encodeURIComponent(type);
-  } else if (index != null && type != null) {
-    if (method == null) method = 'PUT';
-    path = '/' + encodeURIComponent(index) + '/' + encodeURIComponent(type) + '/' + '_mappings';
-  } else if (index != null && type != null) {
-    if (method == null) method = 'PUT';
-    path = '/' + encodeURIComponent(index) + '/' + '_mappings' + '/' + encodeURIComponent(type);
+    path = '/' + encodeURIComponent(index) + '/' + encodeURIComponent(type) + '/';
   } else if (index != null) {
     if (method == null) method = 'PUT';
-    path = '/' + encodeURIComponent(index) + '/' + '_mapping';
+    path = '/' + encodeURIComponent(index) + '/';
   } else if (type != null) {
     if (method == null) method = 'PUT';
-    path = '/' + '_mappings' + '/' + encodeURIComponent(type);
-  } else if (index != null) {
-    if (method == null) method = 'PUT';
-    path = '/' + encodeURIComponent(index) + '/' + '_mappings';
+    path = '/' + '_mappings' + '/' + encodeURIComponent(type) + '/';
   } else {
     if (method == null) method = 'PUT';
-    path = '/' + '_mapping' + '/' + encodeURIComponent(type);
+    path = '/' + '_mapping' + '/';
   }
 
+  path += '_mapping';
```

Comment:
This will make `/_maping/_mapping` in the last `else` case at least.

Reply:
`/_mapping/_mapping` shouldn't be the case in any way, right?